packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.java

```diff
@@ -322,440 +154,198 @@ private void launchCameraWithPermissionsGranted(Promise promise, Intent cameraIn
+        libraryIntent.setType("video/*");
+        break;
+      case "All":
+        libraryIntent.setType("*/*");
+        libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*", "video/*"});
+        break;
     }
 
     mPromise = promise;
 
     libraryIntent.setAction(Intent.ACTION_GET_CONTENT);
-    startActivityOnResult(libraryIntent, REQUEST_LAUNCH_IMAGE_LIBRARY, promise);
+    startActivityOnResult(libraryIntent, ImagePickerConstance.REQUEST_LAUNCH_IMAGE_LIBRARY, promise);
   }
 
-  public void onActivityResult(final int requestCode, final int resultCode, final Intent intent) {
-    if (requestCode == CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE) {
-      if (mLaunchedCropper) {
-        mLaunchedCropper = false;
-        final Promise promise = mPromise;
-        mPromise = null;
-        Bundle exifData = mExifData;
-        mExifData = null;
-
-        if (promise == null) {
-          return;
-        }
-        if (resultCode != Activity.RESULT_OK) {
-          Bundle response = new Bundle();
-          response.putBoolean("cancelled", true);
-          promise.resolve(response);
-          return;
-        }
-
-        handleCropperResult(intent, promise, exifData);
-      }
-      return;
+  public void startCropIntent(Uri uri, String type, int requestCode) throws IOException {
+    String extension = ".jpg";
+    Bitmap.CompressFormat compressFormat = Bitmap.CompressFormat.JPEG;
+    if (type.contains("png")) {
+      compressFormat = Bitmap.CompressFormat.PNG;
+      extension = ".png";
+    } else if (type.contains("gif")) {
+      // If we allow editing, the result image won't ever be a GIF as the cropper doesn't support it.
+      // Let's convert to PNG in such case.
+      extension = ".png";
+      compressFormat = Bitmap.CompressFormat.PNG;
+
+    } else if (type.contains("bmp")) {
+      // If we allow editing, the result image won't ever be a BMP as the cropper doesn't support it.
+      // Let's convert to PNG in such case.
+      extension = ".png";
+      compressFormat = Bitmap.CompressFormat.PNG;
+
+    } else if (!type.contains("jpeg")) {
+      Log.w(ImagePickerConstance.TAG, "Image type not supported. Falling back to JPEG instead.");
+      extension = ".jpg";
+    }
+
+    Uri fileUri;
+    if (requestCode == ImagePickerConstance.REQUEST_LAUNCH_CAMERA) {
+      // if the image is created by camera intent we don't need a new path - it's been already saved
+      fileUri = uri;
+    } else {
+      File file = new File(ImagePickerFileUtils.generateOutputPath(mContext.getCacheDir(), "ImagePicker", extension));
+      fileUri = Uri.fromFile(file);
     }
 
-    if (mPromise == null || (mCameraCaptureURI == null && requestCode == REQUEST_LAUNCH_CAMERA)
-        || (requestCode != REQUEST_LAUNCH_CAMERA && requestCode != REQUEST_LAUNCH_IMAGE_LIBRARY)) {
-      return;
+    CropImage.ActivityBuilder cropImage = CropImage.activity(uri);
+    if (mPickerOptions.getForceAspect() != null) {
```

Comment:
`mPickerOptions` seems to be good candidate for method argument instead of field, ain't it?